### PR TITLE
Cudos 432: Refactor `run` command

### PR DESCRIPTION
**Pull request by Holymir**
Refactor `run` command

### CODE REVIEW

File: packages/cudos-cmd/run/run.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+global.getContractFactory = getContractFactory
+global.getContractFromAddress = getContractFromAddress
+
+async function runCmd(argv) {
+  if (!fs.existsSync(`${path.resolve('.')}/${argv.scriptFilePath}`)) {
+    console.log(`Script at location ${path.resolve('.')}/${argv.scriptFilePath} does not exist. Execute cudo run --help for more info.`)
+    return
+  }
+  const ds = new vm.Script(fs.readFileSync(argv.scriptFilePath))
+  return await ds.runInThisContext()
```
Comment (maptuhec) [inline packages/cudos-cmd/run/run.js]: `await` is redundant in `return` statement. Functions are automatically awaited, when returned

File: packages/cudos-config/config.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const VError = require('verror')
+
+let config = {}
+
+const configPath = path.join(process.cwd(), 'cudos.config.js')
+
+async function getConfig() {
+  if (await fsExstra.pathExists(configPath)) {
+    config = require(configPath)
+  } else {
```
Comment (maptuhec) [inline packages/cudos-config/config.js]: can we skip the `else` statement and return in the if?

File: packages/cudos-utilities/command-service.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
+// TODO: remove this file when done refactoring
+
+const fs = require('fs')
```
Comment (maptuhec) [inline packages/cudos-utilities/command-service.js]: are we done with the refactoring so we could delete this file?

File: packages/cudos-utilities/contract-services.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    const uploadReceipt = await this.uploadContract()
+    console.log(uploadReceipt)
+    const ic = await this.initContract(uploadReceipt.codeId)
+    console.log(ic)
+    this.contractAddress = ic.contractAddress
+    return ic.contractAddress
+  }
+
+  async uploadContract() {
+    const uploadFee = calculateFee(1_500_000, this.gasPrice)
```
Comment (maptuhec) [inline packages/cudos-utilities/contract-services.js]: what is this `1_500_00` ?

File: packages/cudos-utilities/contract-services.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
+const {
```
Comment (maptuhec) [inline packages/cudos-utilities/contract-services.js]: I would rename the file to `contract-utils`

File: packages/cudos-utilities/keypair.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+const bip39 = require('bip39')
+const bip32 = require('bip32')
+const secp256k1 = require('secp256k1')
+const {
+  bech32
+} = require('bech32')
+const createHash = require('create-hash')
+
+function createFromMnemonic(mnemonic, hdPath) {
+  if (!mnemonic) {
```
Comment (maptuhec) [inline packages/cudos-utilities/keypair.js]: do we want to create the wallet if the mnemonic is not correnct or missing?

File: packages/cudos-utilities/keypair.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    const mnemonic = bip39.generateMnemonic(256)
+    const privateKey = seedToPrivateKey(mnemonic, hdPath)
+    return {
+      mnemonic: mnemonic,
+      privateKey: privateKey
+    }
+  } else {
```
Comment (maptuhec) [inline packages/cudos-utilities/keypair.js]: do you need the `else` when you are returning in the `if`?

File: packages/cudos-utilities/keystore.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  DirectSecp256k1Wallet
+} = require('cudosjs')
+const fsExtra = require('fs-extra')
+const keypair = require('./keypair')
+
+const {
+  getAccountByName
+} = require('../cudos-config/config')
+
+const FAUCET_MNEMONIC = 'excite identify move harvest grocery flat tank appear multiply early whisper bronze morning giggle pony genius normal priority truly assume false creek pulse twenty'
```
Comment (maptuhec) [inline packages/cudos-utilities/keystore.js]: could we copy this into a config file?

File: packages/cudos-utilities/keystore.js [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  async getSigner(name) {
+    const acc = await this.loadAccount(name)
+    return await DirectSecp256k1Wallet.fromKey(acc.privateKey, this.network)
+  }
+}
+
+const ks = new KeyStore()
+const p = ks.initFaucetAccount()
```
Comment (maptuhec) [inline packages/cudos-utilities/keystore.js]: what is `p`?

**Review by maptuhec — CHANGES_REQUESTED** [review]

File: packages/cudos-utilities/contract-services.js (hunk 5 above)
Comment (aemil145) [reply packages/cudos-utilities/contract-services.js]: isn't it the full name `contract-utilities` better? (as in the folder name)

Comment (maptuhec) [reply packages/cudos-utilities/contract-services.js]: for me, no, `utils` is way more common name in my opinion

**Review by aemil145 — APPROVED** [review]

File: packages/cudos-utilities/contract-utils.js [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+      try {
+        this.wasmPath = path.join(process.cwd(), `artifacts/${this.contractname}.wasm`)
+      } catch (ex) {
+        console.error(`Contract with name ${this.contractname} was not found, did you compile it ? \n run cudo --help for more available commands`)
+      }
+    }
+
+    const gasPrice = await getGasPrice()
+
+    if (!getGasPrice()) {
```
Comment (maptuhec) [inline packages/cudos-utilities/contract-utils.js]: you don't need this check, becuase you are checking for the same in the config? even if you have the check why dont you check the value of the constant on row 44?

**Review by maptuhec — CHANGES_REQUESTED** [review]

**Review by maptuhec — APPROVED** [review]

**Pull request merged**